File: front/src/api/message.ts
import request from './request'

export interface Message {
  id: number
  conversationId: string
  agentId: string
  type: 'system' | 'user' | 'assistant' | 'tool'
  content: string
  toolCallId: string | null
  toolName: string | null
  toolCallsJson: string | null
  sequence: number
  metadataJson: string | null
  createdAt: string
}

export interface CreateConversationRequest {
  agentId: string
  systemPrompt?: string
}

export interface SendMessageRequest {
  agentId: string
  content: string
}

export interface GraphExecution {
  id: number
  threadId: string
  conversationId: string
  agentId: string
  status: string
  currentNode: string
  progress: number
  errorMessage: string | null
  requirementDocPath: string | null
  planJson: string | null
  extraData: string | null
  createdAt: string
  updatedAt: string
}

// 获取会话列表（指定 Agent）
export function listConversations(agentId: string) {
  return request.get<any, string[]>('/messages/conversations', {
    params: { agentId },
  })
}

// 创建新会话
export function createConversation(data: CreateConversationRequest) {
  return request.post<any, { conversationId: string }>('/messages/conversations', data)
}

// 获取会话消息列表
export function getMessages(conversationId: string) {
  return request.get<any, Message[]>(`/messages/conversations/${conversationId}`)
}

// 获取会话最近 N 条消息
export function getRecentMessages(conversationId: string, limit: number = 50) {
  return request.get<any, Message[]>(`/messages/conversations/${conversationId}/recent`, {
    params: { limit },
  })
}

// 发送消息
export function sendMessage(conversationId: string, data: SendMessageRequest) {
  return request.post<any, Message>(`/messages/conversations/${conversationId}/send`, data)
}

// 删除会话
export function deleteConversation(conversationId: string) {
  return request.delete(`/messages/conversations/${conversationId}`)
}

// 获取会话消息数量
export function countMessages(conversationId: string) {
  return request.get<any, { count: number }>(`/messages/conversations/${conversationId}/count`)
}

// 获取会话最新的 Graph 执行状态
export function getGraphStatus(conversationId: string) {
  return request.get<any, GraphExecution | null>(`/messages/conversations/${conversationId}/graph-status`)
}

// 获取会话的需求文档内容
export function getRequirementDoc(conversationId: string) {
  return request.get<any, { content: string } | null>(`/messages/conversations/${conversationId}/requirement-doc`)
}

// 更新会话的需求文档内容
export function updateRequirementDoc(conversationId: string, content: string) {
  return request.put<any, void>(`/messages/conversations/${conversationId}/requirement-doc`, { content })
}

// 更新会话的计划 JSON
export function updatePlanJson(threadId: string, planJson: string) {
  return request.put<any, void>(`/graph-executions/${threadId}/plan`, { planJson })
}

File: src/main/java/com/fateironist/jawf/model/Message.java
package com.fateironist.jawf.model;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 聊天消息实体类。
 * <p>
 * 用于持久化存储聊天历史，支持与 Spring AI 消息类型的相互转换。
 * <p>
 * <b>消息类型</b>：
 * <ul>
 *   <li>{@code system} - 系统提示词</li>
 *   <li>{@code user} - 用户消息</li>
 *   <li>{@code assistant} - 助手回复</li>
 *   <li>{@code tool} - 工具调用结果</li>
 * </ul>
 */
@Data
public class Message {

    /** 自增主键 */
    private Long id;

    /** 会话 ID（用于关联同一轮对话） */
    private String conversationId;

    /** Agent ID（关联 Agent 表） */
    private String agentId;

    /** 消息类型：system / user / assistant / tool */
    private String type;

    /** 消息内容 */
    private String content;

    /** 工具调用 ID（type=tool 时必填） */
    private String toolCallId;

    /** 工具名称（type=tool 时可选） */
    private String toolName;

    /** 工具调用请求（type=assistant 且包含工具调用时，JSON 格式） */
    private String toolCallsJson;

    /** 序号（同一会话内的消息顺序） */
    private Integer sequence;

    /** 扩展元数据（JSON 格式） */
    private String metadataJson;

    /** 创建时间 */
    private LocalDateTime createdAt;

    /**
     * 消息类型枚举。
     */
    public enum Type {
        SYSTEM("system"),
        USER("user"),
        ASSISTANT("assistant"),
        TOOL("tool");

        private final String value;

        Type(String value) {
            this.value = value;
        }

        public String getValue() {
            return value;
        }

        public static Type fromValue(String value) {
            for (Type type : values()) {
                if (type.value.equals(value)) {
                    return type;
                }
            }
            throw new IllegalArgumentException("未知的消息类型: " + value);
        }
    }
}
